x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -400,13 +400,8 @@ const updateExistingIndex = async ({
   delete mappings.properties.stream;
   delete mappings.properties.data_stream;
 
-  // get the data_stream values from the index template to compose data stream name
-  const indexMappings = await getIndexMappings(indexName, callCluster);
-  const dataStream = indexMappings[indexName].mappings.properties.data_stream.properties;
-  if (!dataStream.type.value || !dataStream.dataset.value || !dataStream.namespace.value)
-    throw new Error(`data_stream values are missing from the index template ${indexName}`);
-  const dataStreamName = `${dataStream.type.value}-${dataStream.dataset.value}-${dataStream.namespace.value}`;
-
+  const [, dsType, dsTemplateName, dsNamespace] = indexName.split('-');
```

Comment:
If I understand correctly `indexName` is a only a `string` vs some other type. Can we at least add some assertions to guard against `undefined` values or anything to give us more confidence it's what we expect?

Reply:
Yeah, it's probably worth checking. If the backing index prefix changes so something like `.ds_logs-...` we'd also have an invalid type. Maybe we should check that the `type` is either `logs` or `metrics`? I'm not sure if it's valid for it to be anything else. Or maybe we can use the enums that we use to create the kibana index patterns and make sure it at least is one of those that way we'd only have to change a single place to add new valid types.